### PR TITLE
`Table` TypeScript migration - `Tr` component [04]

### PR DESCRIPTION
**Pull request by curthashicorp**
### :pushpin: Summary

Feature branch: https://github.com/hashicorp/design-system/pull/2297

`Table::Tr` component TS Migration

### :camera_flash: Screenshots

tr

### :link: External links

Jira ticket: [HDS-3603](https://hashicorp.atlassian.net/browse/HDS-3603)

***

### 👀 Component checklist

- [x] Percy was checked for any visual regression
- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

[HDS-3598]: https://hashicorp.atlassian.net/browse/HDS-3598?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[HDS-3599]: https://hashicorp.atlassian.net/browse/HDS-3599?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[HDS-3600]: https://hashicorp.atlassian.net/browse/HDS-3600?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[HDS-3601]: https://hashicorp.atlassian.net/browse/HDS-3601?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[HDS-3603]: https://hashicorp.atlassian.net/browse/HDS-3603?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/src/components/hds/table/tr.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import type { HdsTableScope } from './types';
+import { HdsTableScope as HdsTableScopeValues } from './types.ts';
+import type { HdsFormCheckboxBaseSignature } from '../form/checkbox/base';
+
+export interface BaseHdsTableTrArgs {
+  Args: {
+    isSelectable?: boolean;
+    isSelected?: false;
+    selectionAriaLabelSuffix?: string;
+    selectionKey?: string | undefined;
```
Comment (alex-ju) [inline packages/components/src/components/hds/table/tr.ts:17]: question: doesn't the `?` cover for `undefined`?
```suggestion
    selectionKey?: string;
```

File: packages/components/src/components/hds/table/tr.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    selectionScope: HdsTableScope;
+    didInsert: (
+      checkbox: HdsFormCheckboxBaseSignature['Element'],
+      selectionKey: string | undefined
```
Comment (alex-ju) [inline packages/components/src/components/hds/table/tr.ts:21]: same question here
```suggestion
      selectionKey?: string
```

**Review by alex-ju — APPROVED** [review]

File: packages/components/src/components/hds/table/tr.ts (hunk 1 above)
Comment (curthashicorp) [reply packages/components/src/components/hds/table/tr.ts:17]: Yes it does. Good catch.

File: packages/components/src/components/hds/table/tr.ts (hunk 2 above)
Comment (curthashicorp) [reply packages/components/src/components/hds/table/tr.ts:21]: Yes it does. Good catch.

**Pull request merged**